src/Namshi/Innovate/Client.php

```diff
@@ -167,6 +181,35 @@ public function createMpiRequest($method = 'GET', $uri = null, $headers = null,
     }
 
     /**
+     * Given a cart id reference will search for transaction for that cart id on innovate
+     *
+     * @param $ref
+     * @return \SimpleXMLElement
+     * @throws InnovateException
+     */
+    public function searchPaymentsByCartId($ref)
+    {
+        $client = new \Guzzle\Http\Client(self::INNOVATE_SEARCH_URL, array(
+            'request.options' => array(
+                'auth'    => array($this->merchantId, $this->key, 'Basic'),
+            )
+        ));
+        $request = $client->get('/tools/api/xml/transaction/'.$ref.'/cart');
+        $response = $this->send($request);
+
+        if ( ! $response instanceof \Guzzle\Http\Message\Response) {
+            throw new InnovateException('Error while connecting to innovate: ' . $response->__toString());
```

Comment:
Should we give the request/response objects as parameters of the InnovateException?